repair-tool.lic

```diff
@@ -0,0 +1,108 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#enchant
+=end
+
+custom_require.call(%w[common common-items events common-crafting equipmanager])
+
+class RepairTool
+  include DRC
+  include DRCC
+  include DRCA
+  include DRCI
+
+  def initialize
+    @settings = get_settings
+    @hometown = @settings.hometown
+    @bag = @settings.crafting_container
+
+    # not used, but passed to stow_crafting_item in crafting-common.lic
+    @belt = @settings.crafting_container
+    # not used, but passed to stow_crafting_item in crafting-common.lic
+    @bag_items = @settings.crafting_items_in_container
+
+    crafting_data = get_data('crafting')
+    @recipe_parts = crafting_data['recipe_parts']
+    @discipline = 'blacksmithing'
+    info = crafting_data[@discipline][@hometown]
+    unless info
+      echo("No crafting settings found for discipline: #{discipline}")
+      exit
+    end
+
+    @brush = @settings.repair_tools.find { |item| /brush/ =~ item } || 'brush'
+    @oil = @settings.repair_tools.find { |item| /oil/ =~ item } || 'oil'
+
+    @brush_part_name = "wire brush"
+    @oil_part_name = "flask of oil"
+
+    arg_definitions = [
+      [
+        { name: 'tool', display: 'tool name', regex: /^[A-z\s\-']+$/i, variable: true, description: 'Name of the tool, wrap in double quotes if this is multiple words.' },
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
```

Comment:
I usually see the args being parsed at the top of the `initialize` function. Also makes it easier to use the `args` later without needing to move as much code around.

Reply:
I prefer to parse args first thing in case there is a problem - we report that to the user ASAP without doing a bunch of other stuff that turned out to be unnecessary.